pkg/azuremetadata/azuremetadata.go

```diff
@@ -0,0 +1,127 @@
+package azuremetadata
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/giantswarm/microerror"
+)
+
+const (
+	infoEndpoint   = "http://169.254.169.254/metadata/instance?api-version=2019-08-01"
+	eventsEndpoint = "http://169.254.169.254/metadata/scheduledevents?api-version=2019-08-01"
+
+	ackEventBody = `{
+	"StartRequests" : [
+		{
+			"EventId": %s
+		}
+	]
+}`
+)
+
+type Client struct {
+	httpClient          *http.Client
+	localInstanceVMName string
+}
+
+func New() (*Client, error) {
```

Comment:
If we pass the `httpClient` as parameter to `New()`, it would be easier to test later on

Reply:
added an optional config for the http client